src/NuGet.Services.Staging.Common/KeyVaultConfigurationReader.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Primitives;
+using NuGet.Services.KeyVault;
+
+namespace NuGet.Services.Staging.Common
+{
+    public class KeyVaultConfigurationReader : IConfigurationRoot
```

Comment:
>KeyVaultConfigurationReader [](start = 17, length = 27)

in the future (hopefully) we will have all our services & jobs using Microsoft.Extensions.Configuration, and we will want them all to also use this code. So lets make it more general:
1. Change the assembly name to something common: NuGet.Services.Configuration.SecretReader
2. Change the name of the argument u use, because "KeyVault:VaultName" is not a valid name in a cloud service configuration because of the ":"